lib/chef/resource/openssl_x509_certificate.rb

```diff
@@ -212,13 +212,18 @@ def request
         end
 
         def subject
+          if new_resource.common_name.nil? && new_resource.subject_alt_name.empty?
+            Chef::Log.fatal("Neither common_name nor subject_alt_name specified, one is required.")
+            raise "Neither common_name nor subject_alt_name specified, one is required."
```

Comment:
While this may be the correct way of doing things, I have concerns that this could break people by surprise. Can we do a deprecation first?

Reply:
@tpowell-progress that makes sense, thanks. I'm new to deprecation warnings in chef resources, is this a good pattern to follow? https://github.com/chef/chef/blob/d06dc07c825f516752b9b26e443e9c6da38a9435/lib/chef/resource/archive_file.rb#L137

Reply:
Looking around, I think that pattern makes sense, provided that you can accomplish the deprecation warning without getting too messy. If for some reason affixing the deprecation warning doesn't look clean, we can dig deeper, but I'd recommend just pushing a reasonable try at it and we can see if makes sense.